### ISSUE TITLE
add ability to change text color and/or change text color automatically

### ISSUE THREAD

**Issue by nnmrts** [badgen/badgen#67]
Feel free to split this issue into two seperate ones, once it's needed.

shields.io kinda does this already. Choosing a `color` of `#ffffff` for example, results in dark text instead of the default white one:

![](https://img.shields.io/badge/badge-test-ffffff)
https://img.shields.io/badge/badge-test-ffffff

badgen does this:

![](https://badgen.net/badge/badge/test/ffffff)
https://badgen.net/badge/badge/test/ffffff

I think shields.io uses a contrast checker internally, probably https://github.com/jxnblk/colorable to determine if it should use bright or dark text. While badgen should do the same, I think an option to customize the text color would be great as well. This is something shields.io doesn't have btw.

Being able to change icon color like you can with shields.io would be nice too, but I guess that's yet another seperate issue.

**Comment by nnmrts** [comment]
@amio ? Is this repo still maintained?

**Comment by amio** [comment]
@nnmrts Yep, sorry for the late response, just a little busy recently. PR's welcome 👋
